gcovr/coverage.py

```diff
@@ -100,20 +109,23 @@ def key_percent_uncovered(key: str) -> float:
         covered = stat.covered
         total = stat.total
 
-        if total > 0:
-            return covered / total
-        return -1 if sort_reverse else 1e99
+        # No branches are always put directly after (or before when reversed)
+        # files with 100% coverage (by assigning such files 110% coverage)
+        return covered / total if total > 0 else 1.1
 
     if sort_key == "uncovered-number":
         key_fn = key_num_uncovered
     elif sort_key == "uncovered-percent":
         key_fn = key_percent_uncovered
     else:
-        # by default, we sort by filename alphabetically
+        # By default, we sort by filename alphabetically
         return sorted(covdata, key=key_filename, reverse=sort_reverse)
 
-    # First sort filename alphabetical and then by the requested key
-    return sorted(sorted(covdata, key=key_filename), key=key_fn, reverse=sort_reverse)
+    return sorted(
+        covdata, key=lambda key: (
+            key_fn(key), key_num_lines(key), key_filename(key)
+        ), reverse=sort_reverse
+    )
```

Comment:
Using a single sort also reverse the alphabetical order. If I'm sorting reverse by branch coverage is it required to sort also the filenames in reverse order? I don't think so.

Reply:
I was on the fence about this one, thinking that reversing the sorting should mirror it, but I do agree with you on this one, it should only be applied to the primary key.